### DISCUSSION TITLE
DB picker does not show active database

### DISCUSSION THREAD

**Discussion by Andreas-E-P** [dbeaver/dbeaver#35822] [Q&A] [answered]
### Description

When I switch to a different database using the `use` statement the DB picker does not switch and still shows the previously active database. I remember that this has been working in the past.
![image](https://github.com/user-attachments/assets/45358287-b9be-4883-b625-c57b4e0a36ac)

### DBeaver Version

24.2.1

### Operating System

Windows 11

### Database and driver

MariaDB with JDBC driver

### Steps to reproduce

1. Run a `use` statement to switch to another database 
2. Run `select database()` to see if the active database really changed to the one from the `use` statement
3. Look at the [DB picker](https://github.com/user-attachments/assets/b9b6a1ac-80c0-48df-9d0b-2977c6902135): The DB that's shown there does not correspond to the one from the `use` statement

[Illustration](https://github.com/user-attachments/assets/45358287-b9be-4883-b625-c57b4e0a36ac)

**Comment by E1izabeth** [comment, marked as answer]
@Andreas-E-P you can turn on the setting `Refresh active schema after SQL execution` to make it work
![image](https://github.com/user-attachments/assets/3ebf1bb0-fd0f-474f-86cc-651fd5ffa420)

**Comment by Andreas-E-P** [comment]
@E1izabeth  
Thank you very much, this solved the issue for me 🙏 🙏   
I really tried hard to find an answer for this on the web before. Weird that I couldn't find anything.  ¯\\_(ツ)_/¯
  
Anyway, I'm really glad that this works now, it was rather annoying to always run "select database()" before doing anything to make sure I'm doing it on the correct database.

**Comment by ask9** [comment]
@E1izabeth 
Shouldn't it be set by default?
If not for the entire application (all database engines), then maybe for some certain database engines supporting switching current database (Sql server and mariaDB are among them, but not Postgres).

**Comment by E1izabeth** [comment]
We changed the default value of this setting intentionally. With this setting, DBeaver sends an additional query after each query execution. We decided that we shouldn't do it by default.
